components/net/http_cache.rs

```diff
@@ -474,11 +498,22 @@ fn handle_range_request(
                 } else {
                     continue;
                 };
-                if res_beginning < beginning && res_end == total - 1 {
+                let total_minus_one = {
```

Comment:
If we check that total != 0 earlier then we don't need a checked subtraction here.